tests/unit-tests/core/main-class.php

```diff
@@ -75,4 +75,17 @@ public function test_wc_class_instances() {
 		$this->assertInstanceOf( 'WC_Payment_Gateways', $this->wc->payment_gateways() );
 		$this->assertInstanceOf( 'WC_Checkout', $this->wc->checkout() );
 	}
+	
+	/**
+	 * Test: user_agent_header
+	 */
+	public function test_user_agent_header() {
+		$expected = 'WooCommerce/3.5.3 (compatible; ClassicCommerce/1.0.0)';
+		$actual = $this->wc->user_agent_header();
+		$expected_with_hook = 'WooCommerce/3.5.3 Hookshot (compatible; ClassicCommerce/1.0.0)';
+		$actual_with_hook = $this->wc->user_agent_header( 'Hookshot' );
```

Comment:
This looks good, the only thing I'd change is to avoid hard-coding the version numbers. For example:

```php
$wc_version = $this->wc->version;
$cc_version = $this->wc->cc_version;
$expected = "WooCommerce/$wc_version (compatible; ClassicCommerce/$cc_version)";
$actual = $this->wc->user_agent_header();
$expected_with_hook = "WooCommerce/$wc_version Hookshot (compatible; ClassicCommerce/$cc_version)";
$actual_with_hook = $this->wc->user_agent_header( 'Hookshot' );
```

Reply:
Yes, agree. I'll change that shortly. My problem is that I've no means of testing these locally which is a right pain so that's why there's so many updates and why I've been keeping things simple like hardcoding things.

Can't get phpunit to work on my Linux box.